benchmarks/stress.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Hugh Ma <Hugh.Ma@flextronics.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: stress
+short_description: Execute the Stress benchmark
+description:
+ - Use this module to run the Stress benchmark
+ - https://people.seas.harvard.edu/~apw/stress/
+version_added: "2.3"
+options:
+  chdir:
+    description:
+     - change working directory
+    required: false
+    default: null
+  cpu:
+    description:
+     - number of cpus to stress
+    required: true
+  delay:
+    description:
+     - piggybacks off the at module as a action plugin to run benchmark at a schedule time
+    required: false
+  executable:
+    description:
+     - path to stress executable if running from source
+    required: false
+  dest:
+    description:
+     - absolute path of file to write stdout to
+    required: false
+  timeout:
+    description:
+     - Total time to run stress for.
+    required: True
+requirements:
+ - stress
+author: "Hugh Ma <Hugh.Ma@flextronics.com>"
+'''
+
+EXAMPLES = '''
+# Run stress for 10 seconds on 1 cpu
+- stress: timeout=10 cpu=1
```

Comment:
Can you use ansible native YAML syntax here ? it considered best practice for new modules.